scripts/test.sh

```diff
@@ -1,3 +1,9 @@
 #!/bin/bash
 
+if ! command -v setup-envtest; then
+	echo "setup-env is required for tests"
+	exit -1
+fi
+
+export KUBEBUILDER_ASSETS=$(setup-envtest use -p path)
```

Comment:
```suggestion
minor=$(go list -m all | grep k8s.io/client-go | cut -d' ' -f2 | cut -d. -f2)
version="1.$minor.x"
export KUBEBUILDER_ASSETS=$(setup-envtest use -p path "$version")
```

That way we test for a specific version so that in the future when an older branch runs the test it doesn't test with an unsupported version of Rancher. WDYT?

Reply:
I'm not sure about that. Per the versioning policy for both wrangler and lasso( [lasso](https://github.com/rancher/rancher/issues/46901), [wrangler](https://github.com/rancher/rancher/issues/46900)), this should support a range of k8s versions, not a single version as determined by client-go. So if we were going to do something like this, ideally this should iterate over a range of versions, run setup with the version, and then run the test suite. However, since the versioning system isn't integrated yet, I'm not sure that this would be something that we want to integrate.

I think what you proposed here is probably a bit more accurate, so I can implement it if we want. However, I think that the lifespan is limited an eventually we will want to remove the go mod scraping in favor of the approach that I proposed above, so I'm not sure that it's worth adding right now. WDYT?

Reply:
I also considered running tests against multiple supported version but figured that would be too much work for a first try, we don't have to get everything right the first time. That said I still think using the minor from client-go is a step in the right direction and it will also prevent CI breakage (eg: "yesterday the CI worked but today it doesn't because upstream released a new version unrelated to this PR's change")

Reply:
Alright, on the basis of reducing flakiness I will add it.